src/input.rs

```diff
@@ -433,25 +433,28 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
-    fn scroll_terminal(&mut self, mouse_modes: TermMode, code: u8) {
+    fn scroll_terminal(&mut self, mouse_modes: TermMode, code: u8, scroll_multiplier: usize) {
         debug_assert!(code == 64 || code == 65);
 
         let faux_scrollback_lines = self.mouse_config.faux_scrollback_lines;
         if self.ctx.terminal_mode().intersects(mouse_modes) {
             self.mouse_report(code, ElementState::Pressed);
-        // } else if faux_scrollback_lines > 0 {
-        //     // Faux scrolling
-        //     let cmd = code + 1; // 64 + 1 = A, 65 + 1 = B
-        //     let mut content = Vec::with_capacity(faux_scrollback_lines * 3);
-        //     for _ in 0..faux_scrollback_lines {
-        //         content.push(0x1b);
-        //         content.push(b'O');
-        //         content.push(cmd);
-        //     }
-        //     self.ctx.write_to_pty(content);
-        // }
+        } else if self.ctx.terminal_mode().contains(TermMode::ALT_SCREEN)
+            && faux_scrollback_lines > 0
```

Comment:
Shouldn't this check whether there's a native buffer **and** whether faux scrolling is enabled?

Asking based on your comment on the original ticket:

> Whenever a native scrollback buffer is available, faux scrollback does not exist.

Reply:
`faux_scrollback_lines` checks if faux scrolling is enabled in the config.
`contains(ALT_SCREEN)` checks if it's in the alternative buffer.
`&&` -> and.

So if scrollback is enabled **and** it's in the alternative buffer. Am I missing something?

Reply:
Does the config override faux_scrollback_lines to be zero when the native buffer is enabled? I just don't see how this will skip the faux scrollback logic when there's a native buffer.

Reply:
What do you mean by "native buffer" exactly? As far as I understand it scrolling in the "normal" way does not exist in the alt screen buffer. So the "native buffer" is just the scroll history that is available inside the main screen buffer, right? By checking for alt buffer it should make sure that the user is not in the main screen buffer which contains the native scroll history.

Reply:
Ah, right. Please see #1000 and discussion in #1147. The TLDR in this case is that the alt screen can have scrollback.

Reply:
I'm not quite sure how this would interact with faux scrolling. In `termite` when inside of `tmux` there is no buffer, the faux scrolling just takes over. The only way to disable that is by disabling faux scrolling.

This behavior seems a bit weird to me, isn't the whole concept of the alternative screen buffer that there is no normal history?

```
The Alternate Screen Buffer is exactly as large
as the display, contains no additional saved lines.
When the Alternate Screen Buffer is active, you cannot
scroll back to view saved lines.
```
 - http://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-The-Alternate-Screen-Buffer

I'm not quite sure if this actually is a "feature". It seems a bit odd to me.

Reply:
Termite, which also supports faux scrolling behaves the same way as this PR.

Reply:
Interesting.. can you also test with urxvt? I won’t have access to a Linux machine until later today.

Reply:
Urxvt supports neither faux scrolling nor native scrolling inside the alternate screen buffer. The same goes for `xterm`.

This seems like a conflict between two features that are not part of any official specification.

The faux scrolling seems to be a "convenient hack" which can make usage easier for some people I guess.

Scrolling inside the alternate screen buffer I haven't seen before, but not only does it look like it's unspecified, it seems like it actually goes against the specification.

@maximbaz do you know any other terminal emulator that does this?

Reply:
I do not, but to be fair I only used urxvt and sakura before I discovered alacritty.

I just tested in sakura, it prefers "tmux history" over faux scroll, in other words when I run `man man` inside tmux, mouse wheel scrolls the scrollback history of tmux, but doesn't send up/down to `man man`. So it does look like we have to choose one or another.

Reply:
Just as an idea, how about keeping both behaviors: when you just scroll the mouse wheel, we do the "faux scroll", but when you hold Shift and scroll the mouse wheel, we do the "alt screen buffer scroll"?

Reply:
Hmm if you prefer one over the other you can already disable faux scrolling in the config. But I like the idea of using the shift key to disable faux scrolling. Would be simple to implement too.

What do you think about the idea @jwilm? I don't think anyone would ever need faux scrolling while holding shift.

Reply:
I've added @maximbaz proposal to this PR for now, allowing the scrolling as long as `shift` is pressed, even with faux scrolling is enabled. I think that's a great workaround and should make it possable for alacritty to support both of these unspecified features without forcing one on anyone.